File: RetailStation.Angular/src/app/components/Merchant/models/MerchantDeliveryRegionModel.ts
import { CreatorModel } from "../../Shared/models/CreatorModel";

export interface MerchantDeliveryRegionModel extends CreatorModel {
    merchantDeliveryRegionId: number | null;
    merchantId: number | null;
    countryId: number;
    cityId: number | null;
    regionId: number | null;
    cost: number;
    deliveryTime: number;
    deliveryTimeUnit: string;
    isActive: boolean;
    countryNameAR: string;
    countryNameEN: string;
    cityNameAR: string;
    cityNameEN: string;
    regionNameAR: string;
    regionNameEN: string;
    totalCount: number | null;
}

File: RetailStation.Entities/DTOs/Operation/MerchantDeliveryRegionModel.cs
using RetailStation.Entities.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RetailStation.Entities.DTOs.Operation
{
    public class MerchantDeliveryRegionModel : CreatorModel
    {
        public int? MerchantDeliveryRegionId { get; set; }
        public int? MerchantId { get; set; }
        public int CountryId { get; set; }
        public int? CityId { get; set; }
        public int? RegionId { get; set; }
        public decimal Cost { get; set; }
        public int DeliveryTime { get; set; }
        public string DeliveryTimeUnit { get; set; }
        public bool IsActive { get; set; } = true;
        public string CountryNameAR { get; set; }
        public string CountryNameEN { get; set; }
        public string CityNameAR { get; set; }
        public string CityNameEN { get; set; }
        public string RegionNameAR { get; set; }
        public string RegionNameEN { get; set; }
        public int? TotalCount { get; set; }


    }
}
